packages/typespec-test/test/openai_generic/generated/typespec-ts/src/api/chat/completions/index.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  chatCompletionRequestMessageSerializer,
+  chatCompletionFunctionsSerializer,
+  CreateChatCompletionRequest,
+  CreateChatCompletionResponse,
+} from "../../../models/models.js";
+import {
+  ChatCompletionsCreate200Response,
+  ChatCompletionsCreateDefaultResponse,
+  isUnexpected,
+  OpenAIContext as Client,
+} from "../../../rest/index.js";
+import {
+  StreamableMethod,
+  operationOptionsToRequestParameters,
+  createRestError,
+} from "@azure-rest/core-client";
+import { serializeRecord } from "../../../helpers/serializerHelpers.js";
+import { ChatCompletionsCreateOptionalParams } from "../../../models/options.js";
+
+export function _createSend(
+  context: Client,
+  body: CreateChatCompletionRequest,
+  options: ChatCompletionsCreateOptionalParams = { requestOptions: {} },
+): StreamableMethod<
+  ChatCompletionsCreate200Response | ChatCompletionsCreateDefaultResponse
+> {
+  return context
+    .path("/chat/completions")
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      body: {
+        model: body["model"],
+        messages: body["messages"].map(chatCompletionRequestMessageSerializer),
+        functions:
+          body["functions"] === undefined
+            ? body["functions"]
+            : body["functions"].map(chatCompletionFunctionsSerializer),
+        function_call: body["functionCall"],
+        temperature: body["temperature"],
+        top_p: body["topP"],
+        n: body["n"],
+        max_tokens: body["maxTokens"],
+        stop: body["stop"],
+        presence_penalty: body["presencePenalty"],
+        frequency_penalty: body["frequencyPenalty"],
+        logit_bias: !body.logitBias
+          ? body.logitBias
+          : (serializeRecord(body.logitBias as any) as any),
+        user: body["user"],
+        stream: body["stream"],
```

Comment:
in this case, is it an anonymous model? if no, should we just generate the body type serializer?

Reply:
Yeah this is getting turned into an anonymous model by the compiler. Will look in a follow up PR if we want to generate models and serializers for anonymous models leveraging TCGCs provided names. For now keeping this like it is